modules/core/src/main/java/org/apache/ignite/internal/managers/communication/transmit/chunk/AbstractChunkedStream.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.managers.communication.transmit.chunk;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicLong;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.managers.communication.transmit.channel.TransmitInputChannel;
+import org.apache.ignite.internal.managers.communication.transmit.channel.TransmitMeta;
+import org.apache.ignite.internal.managers.communication.transmit.channel.TransmitOutputChannel;
+import org.apache.ignite.internal.util.tostring.GridToStringInclude;
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ * The chunked stream with the basic behaviour.
+ */
+abstract class AbstractChunkedStream implements ChunkedInputStream, ChunkedOutputStream {
+    /** Additional stream params. */
+    @GridToStringInclude
+    private final Map<String, Serializable> params = new HashMap<>();
+
+    /** The number of bytes successfully transferred druring iteration. */
+    @GridToStringInclude
+    protected final AtomicLong transferred = new AtomicLong();
+
+    /** The size of segment for the read. */
+    private int chunkSize;
+
+    /** The unique input name to identify particular transfer part.*/
+    private String name;
+
+    /**
+     * The position from which the transfer will start. For the {@link File} it will be offset
+     * where the transfer begin data transfer.
+     */
+    private Long startPos;
+
+    /** The total number of bytes to send. */
+    private Long count;
```

Comment:
Why we use wrappers instead of primitives?

Reply:
Done.